### PR TITLE
measures, decisions, and objective managers are serializable #454

### PR DESCRIPTION
**Pull request by arnaud-m**
introduce a factory of serializable decisions

### CODE REVIEW

**Comment by arnaud-m** [comment]
The fields mentioned by codacy are useless and could be removed or deprecated, because they only ensure backward-compatibility.

**Comment by cprudhom** [comment]
I'll merge the PR asap and take your remarks into account

File: src/main/java/org/chocosolver/solver/search/strategy/assignments/DecisionOperator.java [hunk 1]
```diff
@@ -63,175 +65,14 @@
...
-            return var.getLB() <= value;
-        }
-
-        @Override
-        public DecisionOperator opposite() {
-            return int_split;
-        }
-    };
-
+    DecisionOperator<IntVar> int_eq = DecisionOperatorFactory.makeIntEq();
```
Comment (cprudhom) [inline src/main/java/org/chocosolver/solver/search/strategy/assignments/DecisionOperator.java]: We should consider calling directly the factory and remove these methods from DecisionOperator

**Comment by cprudhom** [comment]
I let you remove useless fields from DecisionOperator, then tell me and I'll merge the PR.

**Comment by arnaud-m** [comment]
OK, I will remove the fields and commit today.

**Comment by gregy4** [comment]
Removed field are not useless, the code is in release 4.0.1 and not mentioned in release notes. I already talked about it with Charles, deprecated directive should be used. It is waste of time for others to find out why their code cannot be compiled.

**Pull request merged**